src/lib/support/jsontlv/JsonToTlv.cpp

```diff
@@ -216,28 +153,14 @@ CHIP_ERROR ParseJsonName(const std::string name, ElementContext & elementCtx, ui
 
     if (nameFields.size() == 2)
     {
-        VerifyOrReturnError(IsUnsignedInteger(nameFields[0]), CHIP_ERROR_INVALID_ARGUMENT);
-
-        char * endPtr;
-        errno                = 0;
-        unsigned long result = strtoul(nameFields[0].c_str(), &endPtr, 10);
-        VerifyOrReturnError(nameFields[0].c_str() != endPtr, CHIP_ERROR_INVALID_ARGUMENT);
-        VerifyOrReturnError((errno != ERANGE && result <= UINT32_MAX), CHIP_ERROR_INVALID_ARGUMENT);
-
-        tagNumber   = static_cast<uint32_t>(result);
+        auto [_, error_code] = std::from_chars(nameFields[0].data(), nameFields[0].data() + nameFields[0].size(), tagNumber, 10);
```

Comment:
Would be this less magic?

```suggestion
        std::errc error_code = std::from_chars(nameFields[0].data(), nameFields[0].data() + nameFields[0].size(), tagNumber, 10).ec;
```

Reply:
I think it's easier to read with the current method, and is idiomatic C++17. The `.ec` is really hidden, and what does  "ec" even mean...